### PR TITLE
Feat splash image

### PR DESCRIPTION
**Pull request by mixcas**
Add metaboxes to _Home_ page for the splash image.
Scaffold the splash image
Makes the splash image work with srcset using lazysizes and its plugins bgset and fit parent

https://github.com/aFarkas/lazysizes/tree/gh-pages/plugins/bgset
https://github.com/aFarkas/lazysizes/tree/gh-pages/plugins/parent-fit

From parent-fit:

> For this to work properly the physical aspect-ratio of the image candidates need to be calculable. To do so at least one of the image candidates inside the srcset attribute also need to include a h (height) descriptor for at least one image candidate in each srcset.

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -20,6 +20,14 @@
     "isotope": "^3.0.4"
   },
   "overrides": {
+    "lazysizes": {
+      "main": [
+        "plugins/bgset/ls.bgset.min.js",
```
Comment (varffs) [inline bower.json:25]: does this order here matter? as you can see in the built files that these files are added to library.js in this order?

File: lib/functions-filters.php [hunk 2]
```diff
@@ -45,3 +45,37 @@ function add_lazysize_on_srcset($attr) {
...
+// Eg.
+// Array (
+//  [1552] => Array (
+//   [url] => http://localhost:8080/insearchofthetruth-net/wp-content/uploads/2017/08/e8b548419ee0ceaf94dbe9f8ce0a6f2f_original.jpg
+//   [descriptor] => h
+//   [value] => 1552w 873
+//   )
+// )
+//
+// NOTE: This is very hacky but works well and doesn't break anything
```
Comment (varffs) [inline lib/functions-filters.php:65]: are these your notes or from a code snippet somewhere? if so can you link it as another comment

File: front-page.php [hunk 3]
```diff
@@ -1,9 +1,27 @@
 <?php
 get_header();
-?>
 
+// Get splash image
+$splash_images = get_post_meta($post->ID, '_igv_home_splash_images', true);
```
Comment (varffs) [inline front-page.php:5]: This (combined with setting the meta boxes on the home page) requires setting a manual homepage in the reading options. I don't like that as we are using front-page.php which is an override. As there is no other data to manage for this front page I think these values should come from site settings. But this is a data question where opinions vary

File: lib/functions-filters.php [hunk 4]
```diff
@@ -45,3 +45,37 @@ function add_lazysize_on_srcset($attr) {
...
+  $height = $size_array[1];
+
+  // Add the height value to the `value` in the original size
+  // $sources is an array that uses the width of each size as the array index
+  $sources[$width]['value'] = $width . 'w ' . $height;
+
+  // Replace the descriptor for h
+  $sources[$width]['descriptor'] = 'h ';
+
+  return $sources;
```
Comment (varffs) [inline lib/functions-filters.php:79]: I'm a bit unclear what this is aiming to do, but on my local is only adds a height to the first value of the set. That seems incorrect to me.

File: src/styl/responsive/mobile.styl [hunk 5]
```diff
@@ -14,6 +14,14 @@
 
 // ******************************************************
 
+// Splash image
+
+.full-splash-image
+  height: 100vh
+  background-size: cover
```
Comment (varffs) [inline src/styl/responsive/mobile.styl:21]: does `cover` no longer center the image? Without adding centering styles it is left aligning?

**Review by varffs — COMMENTED** [review]
This doesnt seem to be working in practice, but no errors to be found in the console either. Basically only the first image in the src set is loaded in as the lazyloaded background, and this first image is the only one with a height value declared. (also the background positioning seems to be wrong but not sure if related). On resize none of the other declared images in the set are loaded

File: bower.json (hunk 1 above)
Comment (mixcas) [reply bower.json:25]: It does, plugins should go first. as noted [here](https://github.com/aFarkas/lazysizes/#plugins)

File: lib/functions-filters.php (hunk 2 above)
Comment (mixcas) [reply lib/functions-filters.php:65]: These are my notes. This is my code.

File: lib/functions-filters.php (hunk 4 above)
Comment (mixcas) [reply lib/functions-filters.php:79]: That's what is supposed to do.

File: front-page.php (hunk 3 above)
Comment (mixcas) [reply front-page.php:5]: I've stated before that I like this pattern better. What do you think @mikeyrayvon?

File: src/styl/responsive/mobile.styl (hunk 5 above)
Comment (mixcas) [reply src/styl/responsive/mobile.styl:21]: Good question. Idk.  Will add background-position

Comment (mikeyrayvon) [reply src/styl/responsive/mobile.styl:21]: you need `background-position: center` doesnt center with just cover.

**Comment by mixcas** [comment]
@varffs I've updated the PR description.

It is working for me. Loads the proper image size and on resize (upscale) it updates the image to a bigger size.

**Comment by varffs** [comment]
Sorry right you are

**Comment by varffs** [comment]
But reconsider the data approach in light of that very nice theme options update to CMB2. Now very easy to have options panel for front page

**Pull request merged**